.github/workflows/types_check.yml

```diff
@@ -0,0 +1,500 @@
+name: Test types support
+
+on:
+  workflow_dispatch:
+
+concurrency:
+  # Allow only one workflow per any non-trunk branch.
+  group: ${{ github.workflow }}-${{ github.ref_name }}-${{ github.ref_name == 'trunk' && github.sha || 'any-sha' }}
+  cancel-in-progress: true
+
+jobs:
+  setup-matrix:
+    name: Setup strategy matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.setup-matrix.outputs.result }}
+
+    steps:
+      - name: Set up matrix
+        uses: actions/github-script@v7
+        id: setup-matrix
+        with:
+          script: |
+            const matrix = [
+              {
+                name: "Linux x64",
+                runner: "ubuntu-latest",
+                target_os: "linux",
+                target_arch: "x86_64",
+                target_arch_go: "amd64"
+              }
+            ];
+
+            return context.eventName === 'pull_request' ? matrix.slice(0, 1) : matrix;
+  
+  build:
+    name: Build ${{ matrix.target.name }} binaries
+    runs-on: ${{ matrix.target.runner }}
+    needs: setup-matrix
+    env:
+      GOVER: 1.22.0
+      GOOS: ${{ matrix.target.target_os }}
+      GOARCH: ${{ matrix.target.target_arch_go }}
+
+    strategy:
+      matrix:
+        target: ${{ fromJson(needs.setup-matrix.outputs.matrix) }}
```

Comment:
Why not just inline the matrix here instead of the extra step to create a json and then parse it?

Reply:
Matrix can't be dynamic, it has to be calculated before scheduling dependent workflow.
That makes sense if we want this workflow behave same as E2E tests: use  single platform for quick PR runs, and all platforms for manual and on push to trunk. @sgrebnov is that correct?

Reply:
@ewgenius - this is correct, thanks, I used the same approach as we do for other workflows where we have dynamic behavior (quick PR run vs full tests). For this particular case I don't think we will want to run such tests as part of PR so we actually don't need matrix to be dynamic so I've simplified the code. Thanks. /cc @phillipleblanc  - updated.

https://github.com/spiceai/spiceai/pull/1218/commits/4a7a51e1a69fe28f80f58d0d26cb8b6b4c49f411